comms/src/inbound_message_service/comms_msg_handlers.rs

```diff
@@ -180,14 +185,32 @@ where
         .map_err(|e| DispatchError::HandlerError(format!("{}", e)))
 }
 
-fn handler_forward<MType>(_message_context: MessageContext<MType>) -> Result<(), DispatchError>
+fn handler_forward<MType>(message_context: MessageContext<MType>) -> Result<(), DispatchError>
 where
     MType: DispatchableKey,
     MType: Serialize + DeserializeOwned,
 {
-    // TODO: Add logic for message forwarding
-
-    Ok(())
+    // Forward message using appropriate broadcast strategy based on the destination provided in the header
+    let envelope = &message_context.message_envelope;
```

Comment:
message_context is owned by this function - should be able to take the message envelope out and send it without cloning it below

Reply:
Change applied